### DISCUSSION TITLE
Support for OpenTelemetry events

### DISCUSSION THREAD

**Discussion by dvisigalli** [SigNoz/signoz#573] [Q&A]
Is there already a way to see OpenTelemetry events attached to the spans? If not, will this feature be implemented?
Thanks

**Comment by pranay01** [comment]
hi @dvisigalli Can you share more details on what do you mean by events attached to events - do you mean support for error events?

cc: @ankitnayan @makeavish

**Reply by dvisigalli** [reply]
OpenTelemetry events are basically the same as Opentracing logs, they are messages that can be attached to spans
([https://opentelemetry.io/docs/instrumentation/java/manual_instrumentation/#create-spans-with-events])

In Jaeger UI for example, those events (and yes, error events too) can be seen inside span bars, or as a list

![image](https://user-images.githubusercontent.com/79854233/148424068-64db29ab-69a1-4293-aa66-4b3839b4847c.png)

I recently started to try Signoz, and it looks really promising, you guys are doing an amazing job, looking forward to see new versions and new features... so thank you in advance

**Reply by pranay01** [reply]
Thanks @dvisigalli 

Yes, we plan to add support for error events soon. You can track this issue here - https://github.com/SigNoz/signoz/issues/246

![Screenshot 2022-01-06 at 11 45 04 PM](https://user-images.githubusercontent.com/504541/148430828-629e6b05-7622-4153-8262-ae0ea443322e.png)

Regarding custom events - we are not sure on how many people use them and what for? Do you have examples for use cases which you would solve with custom events in spans? Would help us understand the requirements better :)

**Reply by dvisigalli** [reply]
Thanks @pranay01 for the reply!
Regarding custom events, actually i don't have a specific use case in mind, but generally speaking i think they could be used as a sort of log messages, so could be cool and useful to see the list of custom events of each span.
Would be a fast and easy way to read the logs of a distributed application, alternative to the use of log processing tools

**Reply by pranay01** [reply]
Got it :)
